python/lsst/dax/ppdb/bigquery/ppdb_bigquery.py

```diff
@@ -165,6 +165,26 @@ def _generate_manifest(
             compression_format=self.parq_compression,
         )
 
+    @classmethod
+    def _is_empty_chunk(cls, chunk_data: list[ApdbTableData]) -> bool:
+        """Check if all provided table data objects in the replica chunk are
+        empty.
+
+        Parameters
+        ----------
+        chunk_data : `list` [ `ApdbTableData` ]
+            List of table data objects to check.
+
+        Returns
+        -------
+        is_empty : `bool`
+            `True` if all table data objects are empty, `False` otherwise.
+        """
+        for table_data in chunk_data:
+            if len(table_data.rows()) > 0:
+                return False
+        return True
```

Comment:
Maybe shorter:

    return any(len(table_data.rows()) > 0 for table_data in chunk_data)